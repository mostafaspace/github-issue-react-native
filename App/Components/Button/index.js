import React from 'react';
import {Text, ActivityIndicator, TouchableOpacity} from 'react-native';
import Styles from './styles';
import {Colors} from '../../Constants';
import PropTypes from 'prop-types';

export const Button = ({text, onPress, containerStyle, isLoading, disabled, testID}) => {
    return (
        <TouchableOpacity
            style={[Styles.container, containerStyle, (isLoading || disabled) && Styles.disabled]}
            disabled={isLoading || disabled}
            onPress={() => onPress && onPress()}
            testID={testID ? testID : 'Button'}
        >
            {isLoading ? (
                <ActivityIndicator size="small" color={Colors.grey} style={Styles.indicator} />
            ) : (
                <Text style={Styles.text} numberOfLines={1} testID={'ButtonText'}>
                    {text}
                </Text>
            )}
        </TouchableOpacity>
    );
};

Button.propTypes = {
    text: PropTypes.string,
    containerStyle: PropTypes.object,
    isLoading: PropTypes.bool,
    disabled: PropTypes.bool,
    onPress: PropTypes.func,
};
