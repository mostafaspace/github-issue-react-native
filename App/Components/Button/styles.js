import {StyleSheet} from 'react-native';
import {Colors} from '../../Constants';

export default StyleSheet.create({
    container: {
        margin: 16,
        paddingHorizontal: 16,
        paddingVertical: 16,
        borderRadius: 16,
        backgroundColor: Colors.blue,
        alignItems: 'center',
        justifyContent: 'space-evenly',
        flexDirection: 'row',

        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 1,
        },
        shadowOpacity: 0.18,
        shadowRadius: 1.0,

        elevation: 1,
    },
    text: {
        flex: 1,
        color: Colors.white,
        alignSelf: 'center',
        textAlign: 'center',
        paddingHorizontal: 16,
    },
    grayText: {
        color: Colors.grey,
    },
    disabled: {
        backgroundColor: Colors.lightGrey,
    },
    indicator: {
        flex: 1,
    },
});
