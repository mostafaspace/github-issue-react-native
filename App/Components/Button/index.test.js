import React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent, cleanup} from '@testing-library/react-native';
import '@testing-library/jest-native/extend-expect';
import {Button} from '../Button';

afterEach(cleanup)
it('test click and and check text', () => {
    const onPressMock = jest.fn();
    const {getByText} = render(<Button text={'click here'} onPress={onPressMock} />);

    fireEvent.press(getByText('click here'));
    expect(onPressMock).toHaveBeenCalled();
});

it('test to check if disabled', () => {
    const {getByTestId} = render(<Button text={'click here'} disabled />);
    expect(getByTestId('Button')).toBeDisabled();
});

it('Button matches snapshot', () => {
    const onPressMock = jest.fn();
    const tree = renderer.create(<Button text={'click here'} onPress={onPressMock} />).toJSON()
    expect(tree).toMatchSnapshot();
});
