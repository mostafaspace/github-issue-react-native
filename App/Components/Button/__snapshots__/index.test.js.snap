// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#149ADE",
      "borderRadius": 16,
      "elevation": 1,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "margin": 16,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 16,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.18,
      "shadowRadius": 1,
    }
  }
  testID="Button"
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "alignSelf": "center",
        "color": "#ffffff",
        "flex": 1,
        "paddingHorizontal": 16,
        "textAlign": "center",
      }
    }
    testID="ButtonText"
  >
    click here
  </Text>
</View>
`;
